aws/awsutil/strings.go

```diff
@@ -0,0 +1,67 @@
+package awsutil
+
+import (
+	"unicode"
+	"unicode/utf8"
+)
+
+// StringHasPrefixFold tests whether the string s begins with prefix, interpreted as UTF-8 strings,
+// under Unicode case-folding.
+//
+// This implementation is a modified version of the Go Standard Library's strings.EqualFold
+// Copyright 2009 The Go Authors. All rights reserved.
+func StringHasPrefixFold(s, prefix string) bool {
```

Comment:
Why not do something like?
```go 
func StringHasPrefixFold(s, prefix string) bool {
	return len(s) >= len(prefix) && Strings.EqualFold(s[0:len(prefix)],prefix)
}
```

I understand modifying the EqualFold implementation directly would be better than the way I suggest. Curious how/why this is a non trivial improvement for the SDK?